sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormatSuite.scala

```diff
@@ -56,4 +55,22 @@ class ParquetFileFormatSuite extends QueryTest with ParquetTest with SharedSpark
     }.getCause
     assert(exception.getMessage().contains("Could not read footer for file"))
   }
+
+  test("read parquet map invaild key type") {
+    withTempDir { dir =>
+      val spark_session = spark
+      import spark_session.implicits._
+      val df = Seq(Map(Map(1 -> 1) -> 1L)).toDF("map")
+      val exception = intercept[AnalysisException] {
+        df.write.mode("overwrite").parquet(dir.getPath)
+      }
+      assert(exception.getMessage().contains("Parquet data source does not support"))
```

Comment:
I'm not sure about what's your suggestion;
```
scala> val df = Seq(Map(Map(1 -> 1) -> 1L)).toDF("map")
df: org.apache.spark.sql.DataFrame = [map: map<map<int,int>,bigint>]

scala> df.printSchema
root
 |-- map: map (nullable = true)
 |    |-- key: map
 |    |    |-- key: integer
 |    |    |-- value: integer (valueContainsNull = false)
 |    |-- value: long (valueContainsNull = false)


scala> df.write.mode("overwrite").parquet("/tmp/test")

scala>  spark.read.load("/tmp/test").printSchema()
root
 |-- map: map (nullable = true)
 |    |-- key: map
 |    |    |-- key: integer
 |    |    |-- value: integer (valueContainsNull = true)
 |    |-- value: long (valueContainsNull = true)
```
Any issue here?

Reply:
o sorry , I seem to be wrong
This side will make it impossible to read. Let me review it again.

Seq(Seq(Map(Map(1 -> 1) -> StructValue(1L)))).toDF("long1")
root
 |-- long1: array (nullable = true)
 |    |-- element: map (containsNull = true)
 |    |    |-- key: map
 |    |    |    |-- key: integer
 |    |    |    |-- value: integer (valueContainsNull = false)
 |    |    |-- value: struct (valueContainsNull = true)
 |    |    |    |-- l: long (nullable = false)

Reply:
btw, to make reviwers understood easily,  you had better clearly write the PR description, what's an issue, example queries to reproduce the issue, and how-to-fix.

Reply:
thanks